x-pack/plugins/fleet/public/applications/fleet/sections/agents/agent_requirements_page/components/fleet_server_on_prem_instructions.tsx

```diff
@@ -698,7 +698,9 @@ export const OnPremInstructions: React.FC = () => {
     if (!modal) {
       refresh();
     }
-  }, [modal, refresh]);
+    // refresh in dep list causes it being called on every render
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I think the fact that we have to disable this eslint rule in order to fix the bug indicates that the `refresh` function is not being properly memorized. We should fix the bug and ensure that `refresh` is memorized between renders instead of excluding it here to fix the "symptom" of that issue.

Reply:
what I don't get, `refresh` is already wrapped in `useCallback`, that should prevent the infinite loop like it is suggested here: https://stackoverflow.com/questions/62601538/react-useeffect-passing-a-function-in-the-dependency-array

Reply:
Looks like that `useCallback` specifies `refreshOutputs` and `refreshSettings` as its dependencies, which in turn from our `useGetSettings` hook.

https://github.com/elastic/kibana/blob/d80d50bd5ed8384699b2ebfd1d8d8e79531021ae/x-pack/plugins/fleet/public/applications/fleet/sections/agents/agent_requirements_page/components/fleet_server_on_prem_instructions.tsx#L281-L283

https://github.com/elastic/kibana/blob/d80d50bd5ed8384699b2ebfd1d8d8e79531021ae/x-pack/plugins/fleet/public/applications/fleet/sections/agents/agent_requirements_page/components/fleet_server_on_prem_instructions.tsx#L244

https://github.com/elastic/kibana/blob/d80d50bd5ed8384699b2ebfd1d8d8e79531021ae/x-pack/plugins/fleet/public/hooks/use_request/settings.ts#L13-L18

Seems likely that one of these two functions is incorrectly having its identity changed on each render.

Reply:
`refreshOutputs` comes from 

https://github.com/elastic/kibana/blob/e8eb222e2533223c366888b25017535142f758a3/x-pack/plugins/fleet/public/hooks/use_request/outputs.ts#L22-L35

Reply:
yes, I think something is wrong with `refreshOutputs`, if I remove that from `refresh` dependencies, the bug is not happening.
`refreshOutputs` was introduced here:
https://github.com/elastic/kibana/pull/113620/files

Reply:
Myabe refresh should only have `resendRequest` and not the whole outputsRequest as dependency

Reply:
I removed `useMemo` and `useCallback` from `useDefaultOutput`, and this fixes too. 
Is it necessary at all? `resendRequest` is already wrapped in `useCallback`.